backend/src/backend/api.py

```diff
@@ -0,0 +1,120 @@
+from fastapi import FastAPI, Response
+from fastapi.responses import StreamingResponse, JSONResponse
+from fastapi.exceptions import HTTPException
+import cv2
+import time
+
+from backend.managers.camera_manager import CameraManger, CameraNotFoundError
+
+# Create api
+app = FastAPI()
+
+# Create camera manager
+camera_manager = CameraManger()
+
+
+def fps_to_ms(fps: int) -> float:
+    """
+    Convert frames per second to frames per miliseconds (time between each frame)
+    """
+    return (1/fps) * 1000
+
+
+def generate_frames(cap: cv2.VideoCapture, camera_name: str):
+    """
+    A generator that will yield the frames for a given camera
+    """
+    # Capture video frames at specified frame rate
+    time_since_last_frame = 0
+    while camera_manager.camera_is_running(camera_name):
+        # Get the wait time based on fps
+        wait_time = fps_to_ms(camera_manager.get_camera_fps(camera_name))
+
+        # Capture a video frame
+        success, frame = cap.read()
+        if not success:
+            break
+
+        # Check if it's time to send a frame
+        if time.time() * 1000 - time_since_last_frame > wait_time:
+            # Get the encoding quality of camera
+            encoding_params = camera_manager.get_camera_encoding_params(
+                camera_name)
+
+            # Encode frame and convert to byte string
+            ret, buffer = cv2.imencode('.jpg', frame, encoding_params)
+            frame = buffer.tobytes()
+
+            # Yield the current frame
+            yield (b'--frame\r\n'
+                   b'Content-Type: image/jpeg\r\n\r\n' + frame + b'\r\n')
+            time_since_last_frame = time.time() * 1000
+    cap.release()
+
+
+@app.get("/stream/start/{camera_name}")
+async def start_stream(camera_name: str) -> StreamingResponse:
+    """
+    Take in a camera name and start a video stream
+    """
+    # If camera is already streaming, return bad response
+    if camera_manager.camera_is_running(camera_name):
+        raise HTTPException(
+            status_code=400, detail=f"{camera_name} is already streaming")
+    try:
+        # Create video capture object for camera
+        cap = camera_manager.start_video_capture(camera_name)
+        cap.set(cv2.CAP_PROP_BUFFERSIZE, 1)  # Set camera parameters.
+
+    except CameraNotFoundError as e:
+        raise HTTPException(status_code=404, detail=str(e))
+    # Return streaming response
+    return (StreamingResponse(generate_frames(
+        cap, camera_name), media_type="multipart/x-mixed-replace; boundary=frame"))
+
+
+@app.post("/stream/end/{camera_name}")
+async def end_stream(camera_name: str) -> Response:
+    """
+    End a video a stream given a camera name
+    """
+    if camera_manager.camera_is_running(camera_name) == True:
+        camera_manager.end_video_capture(camera_name)
```

Comment:
pep8 violation (use `ruff check` or the Ruff extension for your editor)


```suggestion
    if camera_manager.camera_is_running(camera_name):
```

Reply:
note: this suggestion uses the wrong line number. check line 81 instead :)

Reply:
Fixed in ecfc8ed3f5d6e7cb9d1131578be571ca8cd3a31a